client/src/components/DisplayTable/DisplayTable.js

```diff
@@ -71,25 +100,22 @@ export class DisplayTable extends React.Component {
         },
       }
       */,
-      disable_header_utils,
+      enable_utils,
+      custom_utils,
```

Comment:
I think how I handle `enable_utils` and `custom_utils` can be improved here. Any suggestions welcome

Reply:
I'd suggest dropping `enable_utils` and just having one prop for `util_components`. Each user of `DisplayTable` will need to import and pass in any of the shared utils it wants from `DisplayTableUtils.js` along with its own custom utils. A bit more boilerplate everywhere, but fully flexible and consistent between custom and shared utils.

`DisplayTable` will have to pass some reasonable set of props to the util components it renders. That's mostly for the shared utils (particularly download and copy which want the table-processed data), the custom ones can ignore those props and smuggle in any variables or callbacks from the higher scope where they're defined.

Reply:
At least, that's how I'd do it. Could still have set the default value for `util_components` as the current set of shared components, but anywhere that wants a custom set of utils will need to re-insert the shared utils along with the custom ones it passes in.

Reply:
Indeed that's more consistent and flexible, however, that means utils need to be imported and pass in its own copy/download csv everywhere, as well as calculating its own csv data. I personally don't like the idea of having to do more work everytime you make a new table for features when `DisplayTable` is capable of doing it all

Reply:
how about `util_components`, with shared components as booleans, and just leave a room for custom utils in there, and use that?

or, make separate `shared_utils` that store shared util components like `<DisplayTableDownloadCsv>`, have prop like `util_components` as above. If value is `true`, render what's in `shared_utils`, if it's not boolean, just render it? I know that sounds confusing but it really isn't

Reply:
1) RE: importing and passing in shared utils everywhere. All of the default shared utils can be imported by `DisplayTable` and set as the default prop value for `util_components`. Only instances that want any sort of custom `util_components` will have to import and include whichever of the defaults they still want to keep
  2) RE: having to calculate its own data. Can be avoided if we set the standard that the rendering `DisplayTable` passes some consistent set of props, like display-ready data, in to all of the `util_components` its rendering. Both the shared utils and the custom utils could make use of those but, as commented above, the custom utils can also access anything from the scope they're defined in (meaning they can smuggle in things like callbacks and don't necessarily need to make use of any of the props `DisplayTable` tries to pass)

Reply:
If any of that's unclear still, I could implement my suggested approach pretty quickly to show what I mean.

Reply:
ohhhh okay, this is actually in align with what I was thinking of, basically